agent/engine/execcmd/manager_init_task_linux.go

```diff
@@ -0,0 +1,151 @@
+// +build linux
+
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+package execcmd
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/pborman/uuid"
+
+	apicontainer "github.com/aws/amazon-ecs-agent/agent/api/container"
+	apitask "github.com/aws/amazon-ecs-agent/agent/api/task"
+	taskresourcevolume "github.com/aws/amazon-ecs-agent/agent/taskresource/volume"
+)
+
+const (
+	// TODO: [ecs-exec] decide if this needs to be configurable or put in a specific place in our optimized AMIs
+	ContainerBinDir      = "/usr/bin"
+	BinName              = "amazon-ssm-agent"
+	SessionWorkerBinName = "ssm-session-worker"
+	ConfigFileName       = "amazon-ssm-agent.json"
+
+	// TODO: [ecs-exec] decide if this needs to be configurable or put in a specific place in our optimized AMIs
+	HostLogDir          = "/var/log/ecs/exec"
+	ContainerLogDir     = "/var/log/amazon/ssm"
+	HostCertFile        = "/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem"
+	ContainerCertFile   = "/etc/ssl/certs/ca-certificates.crt"
+	ContainerConfigFile = "/etc/amazon/ssm/amazon-ssm-agent.json"
+
+	namelessContainerPrefix = "nameless-container-"
+	internalNamePrefix      = "internal-execute-command-agent"
+	logVolumeNamePrefix     = internalNamePrefix + "-log"
+	certVolumeName          = internalNamePrefix + "-tls-cert"
+	configVolumeName        = internalNamePrefix + "-config"
+)
+
+// InitializeTask adds the necessary volumes and mount points in all of the task containers in order for the
+// exec cmd agent to run upon container start up.
+// TODO: [ecs-exec] Should we validate the ssm agent binaries & certs are valid and fail here if they're not? (bind mount will succeed even if files don't exist in host)
```

Comment:
Either here, or in our execute-command capability enablement...
or both.  I think if we check before we enable the capability, that will be enough